src/modules/70-pipeline/pages/execution/ExecutionTestView/interfaces/TIUIApp.types.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import type React from 'react'
+import type { ExecutionContextParams } from '@pipeline/context/ExecutionContext'
+import type { CopyTextProps } from '@common/components/CopyText/CopyText'
+import type { DurationProps } from '@common/exports'
+import type {
+  UseExpandErrorModalProps,
+  UseExpandErrorModalReturn
+} from '@pipeline/components/ExpandErrorModal/useExpandErrorModal'
+export interface TIUIAppCustomProps {
+  context: ExecutionContextParams
+  customComponents: {
+    CopyText: React.ComponentType<CopyTextProps>
+    Duration: React.ComponentType<DurationProps>
```

Comment:
Instead of reconstructing a type from props, you can just use the type from the thing itself.
```suggestion
    CopyText: typeof CopyText
    Duration: typeof Duration
```

And then above just import the things using the `type` prefix
```typescript
import type { CopyText } from '@common/components/CopyText/CopyText'
import type { Duration } from '@common/exports'
```

Reply:
this is amazing - thanks!